Fixes for updates in cdrstream and typelib interface

dpotman opened the pull request:
Updates related to some minor interface changes in cdrstream and typelib in DDSI, related to https://github.com/eclipse-cyclonedds/cyclonedds/pull/1468.

eboasson reviewed (approved):
LGTM, but obviously needs to wait until https://github.com/eclipse-cyclonedds/cyclonedds/pull/1468 is merged.

dpotman commented:
I forgot to add some changes in the `xtypes_sub.c` (fuzzer tests), and `dds_cdrstream_desc_from_topic_desc` was missing support for keys, which is now added in https://github.com/eclipse-cyclonedds/cyclonedds/pull/1468 and also needed a few changes in the fuzzer test, so I've added commit [411bc8b](https://github.com/eclipse-cyclonedds/cyclonedds-python/pull/159/commits/411bc8b4e75fc810df15c13a45d29e09e1534a27) that fixes both.
